File: web/python/examples/shared_text.py
"""Typed Python counterpart to noon-native's shared_text example.

Both objects are ordinary members of one Rust semantic Scene. Native Text is
shaped into that same store and the browser boundary transfers only the mixed
resource bundle and retained execution deltas to the renderer.
"""

from noon import Circle, RIGHT, LEFT, Scene, Text


class SharedText(Scene):
    def construct(self):
        circle = Circle(radius=0.5).shift(LEFT * 2)
        label = Text("Noon", font_size=48).shift(RIGHT)
        self.add(circle, label)

File: crates/noon-native/examples/shared_text.rs
//! Typed native execution of one shared semantic scene containing geometry and text.

use noon::{Scene, Text};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut scene = Scene::new();

    let mut circle = scene.circle(0.5)?;
    circle.shift(-2.0, 0.0)?;
    scene.add(&circle)?;

    let mut label = scene.text(Text::new("Noon").with_font_size(48.0))?;
    label.shift(1.0, 0.0)?;
    scene.add(&label)?;

    let session = scene.execution_session()?;
    noon_native::run(session)?;
    Ok(())
}
